File: 2873-maximum-value-of-an-ordered-triplet-i/2873-maximum-value-of-an-ordered-triplet-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumTripletValue = function(nums) {
    let res = 0, diff = 0, left = 0

    for (const i of nums) {
        res = Math.max(res, diff * i)
        diff = Math.max(diff, left - i)
        left = Math.max(left, i)
    }
    return res
    
};

File: 2873-maximum-value-of-an-ordered-triplet-i/2873-maximum-value-of-an-ordered-triplet-i.py
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        # Input: an array of integers nums
        # Output: an integer with the maximum value over all triplets of indices
        
        # Algorithm: Burute Force

        res = 0
        N = len(nums)

        # Iterate through all possible j indices
        for j in range(1, N - 1):
            # Find the maximum value to the left of j
            max_left = 0
            for i in range(j):
                max_left = max(max_left, nums[i])

            # Find the maximum value to the right of j
            max_right = 0
            for k in range(j + 1, N):
                max_right = max(max_right, nums[k])

            # Calculate the triplet value and update res
            value = (max_left - nums[j]) * max_right
            res = max(res, value)

        # Return the final result
        return res if res > 0 else 0
